ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -42,6 +42,19 @@ let compute_memory_overhead ~__context ~vm =
 
 let update_memory_overhead ~__context ~vm = Db.VM.set_memory_overhead ~__context ~self:vm ~value:(compute_memory_overhead ~__context ~vm)
 
+let is_feature_on ~__context ~self ~feature =
+  let vm_gm = Db.VM.get_guest_metrics ~__context ~self in
+  let vm_gmr = try Some (Db.VM_guest_metrics.get_record_internal ~__context ~self:vm_gm) with _ -> None in
+  match vm_gmr with
+  | None -> false
+  | Some vm_gmr ->
+    let other = vm_gmr.Db_actions.vM_guest_metrics_other in
+    try
+      match List.assoc feature other with
```

Comment:
I am uncomfortable with another function that turns strings into booleans unless there is a good argument that this string is special. Could you use for the string interpretation an existing function?